logstash-core/lib/logstash/runner.rb

```diff
@@ -149,8 +149,12 @@ def execute
       return 1
     end
 
-    # Print a warning to STDERR for bad java versions
-    LogStash::Util::JavaVersion.warn_on_bad_java_version
+    # Exit on bad java versions
+    java_version = LogStash::Util::JavaVersion.version
+    if LogStash::Util::JavaVersion.bad_java_version?(java_version)
+      $stderr.puts "Java version 1.8.0 or later is required. (You are running: #{java_version})"
+      return 1
+    end  
```

Comment:
@ph `signal_usage_error` is a clamp method.. Typically that is used for CLI usage issues. Java requirement is more of a fatal error. See similar usages: https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/runner.rb#L147